service-api/src/main/java/greencity/service/NotificationService.java

```diff
@@ -0,0 +1,67 @@
+package greencity.service;
+
+import greencity.dto.notification.NotificationInternalDto;
+import greencity.dto.notification.NotificationDto;
+
+import java.util.List;
+
+public interface NotificationService {
+
+    public interface NotificationService {
+
+        /**
+         * Creates a new notification based on the internal DTO.
+         *
+         * @param dto DTO containing data for creating a notification (type, object, receiver)
+         */
+        void createNewCommentNotification(NewCommentNotificationDto dto);
+        void createNewReplyNotification(NewReplyNotificationDto dto);
+        void createNewLikeNotification(NewLikeNotificationDto dto);
```

Comment:
if they're already taking DTOs could just have one method that takes the base NotificationDto:
```suggestion
        void createNotification(NotificationDto dto);
```